apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java

```diff
@@ -387,13 +402,26 @@ public Blob getBlob(String container, String name, GetOptions options) {
 
    @Override
    public void removeBlob(String container, String name) {
-      // Multipart objects have a manifest which points to subobjects.  Normally
+      // Multipart objects have a manifest which points to subobjects. Normally
       // deleting a object only deletes the manifest, leaving the subobjects.
       // We first try a multipart delete and if that fails since the object is
       // not an MPU we fall back to single-part delete.
-      DeleteStaticLargeObjectResponse response = api.getStaticLargeObjectApi(regionId, container).delete(name);
-      if (!response.status().equals("200 OK")) {
+
+      String objManifest = getManifestInfo(container, name);
+      String sloInfo = getSLOInfo(container, name);
```

Comment:
Previously this code issued one or two DELETEs depending if the name was an SLO or a regular object.  With this commit we issue always issue three RPCs, getWithoutBody, getWithoutBody, and delete with some flags.  How can you optimize this?  Also this fails if the object does not exist.